src/worker/tasks/nonceHealthCheckWorker.ts

```diff
@@ -0,0 +1,176 @@
+      onchainNonce: currentState.onchainNonce,
+      internalNonce: currentState.internalNonce,
+    };
+  });
+
+  const walletHealthResults = await Promise.all(walletHealthPromises);
+  logWalletHealth(walletHealthResults, job);
+}
+
+// Check if a queue is stuck
+async function isQueueStuck(
+  walletAddress: Address,
+  chainId: number,
+): Promise<boolean> {
+  const historicalStates = await getHistoricalNonceStates(
+    walletAddress,
+    chainId,
+    CHECK_PERIODS,
+  );
+
+  // ensure we have enough data to check
+  if (historicalStates.length < CHECK_PERIODS) return false;
+
+  const oldestOnchainNonce = historicalStates[CHECK_PERIODS - 1].onchainNonce;
+
+  // if for every period, the onchain nonce has not changes, and the internal nonce has strictly increased
+  // then the queue is stuck
+  const isStuckForAllPeriods = historicalStates.every((state, index) => {
+    if (index === historicalStates.length - 1) return true; // Last (oldest) state
+
+    const nextState = historicalStates[index + 1];
+    return (
+      state.onchainNonce === oldestOnchainNonce &&
+      state.internalNonce > nextState.internalNonce
+    );
+  });
+
+  return isStuckForAllPeriods;
+}
+
+// Get current nonce state
+async function getCurrentNonceState(
+  walletAddress: Address,
+  chainId: number,
+): Promise<NonceState> {
+  const [onchainNonce, internalNonce] = await Promise.all([
+    getOnchainNonce(chainId, walletAddress),
+    redis.get(lastUsedNonceKey(chainId, walletAddress)),
+  ]);
+
+  return {
+    timestamp: Date.now(),
+    onchainNonce: parseInt(onchainNonce.toString()),
+    internalNonce: parseInt(internalNonce || "0"),
+  };
+}
+
+function nonceHistoryKey(walletAddress: Address, chainId: number) {
+  return `nonceHistory:${chainId}:${walletAddress}`;
+}
+
+// Get historical nonce states
+async function getHistoricalNonceStates(
+  walletAddress: Address,
+  chainId: number,
+  periods: number,
+): Promise<NonceState[]> {
+  const key = nonceHistoryKey(walletAddress, chainId);
+  const historicalStates = await redis.lrange(key, 0, periods - 1);
+  return historicalStates.map((state) => JSON.parse(state));
+}
+
+// Update nonce history
+async function updateNonceHistory(walletAddress: Address, chainId: number) {
+  const currentState = await getCurrentNonceState(walletAddress, chainId);
+  const key = nonceHistoryKey(walletAddress, chainId);
+
+  await redis
+    .multi()
+    .lpush(key, JSON.stringify(currentState))
+    .ltrim(key, 0, CHECK_PERIODS - 1)
+    .exec();
+}
+
+// Log wallet health
+function logWalletHealth(healthResults: WalletHealth[], job: Job<string>) {
+  healthResults.forEach((result) => {
+    const message =
+      `[WALLET_HEALTH] ${result.walletAddress}:${result.chainId} ` +
+      `isStuck:${result.isStuck} ` +
+      `onchainNonce:${result.onchainNonce} ` +
+      `internalNonce:${result.internalNonce}`;
+
+    if (result.isStuck) {
+      logger({
+        service: "worker",
+        level: "warn",
```

Comment:
IMO if a queue is stuck the log level should be something where we would monitor and likely page on-call. I'd make this level `fatal` or `error`.

No need to change the actual message. We can create monitors on the level, so the if statement can just control the level:
```typescript
logger({
  service: "worker",
  level: result.isStuck ? "fatal" : "info",
  message,
})
```